src/main/java/chattime/parser/Parser.java

```diff
@@ -0,0 +1,315 @@
+package chattime.parser;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeParseException;
+import java.util.regex.Pattern;
+
+import chattime.command.AddCommand;
+import chattime.command.ByeCommand;
+import chattime.command.Command;
+import chattime.command.DeleteCommand;
+import chattime.command.FindCommand;
+import chattime.command.ListCommand;
+import chattime.command.MarkCommand;
+import chattime.exception.ChattimeException;
+import chattime.task.Deadline;
+import chattime.task.Event;
+import chattime.task.Task;
+import chattime.task.Todo;
+
+
+/**
+ * Represents parser for bot, processes raw user input and creates appropriate Command object.
+ */
+public class Parser {
+
+    private static final String NO_DESCRIPTION = "OOPS!!! The description of %s cannot be empty.";
+    private static final String MISSED_PARAM = "OOPS!!! %s should be in form of %s.";
+    private static final String NO_INDEX = "OOPS!!! The index to %1$s cannot be empty.";
+    private static final String NEED_INT = "OOPS!!! The index to %1$s must be positive integer.";
+    private static final String IDX_OUT_OF_BOUND = "OOPS!!! The index is too large! We currently have %d task(s).";
+    private static final String UNRECOGNISED_COMMAND = "Sorry... but I don't understand what you said >,<";
```

Comment:
I like the error messages being set as enums, very organised.

Reply:
Yea, I'm finding a way to make things neat :wink: